alephmarcreader/abstractalephmarcreader.py

```diff
@@ -105,6 +105,31 @@ def get_pretty_string(self):
             else:
                 return self.reference
 
+    class StandardizedDate:
+        """
+        Represents a standardized date.
+        Follows the KNORA date format, which can be a single date or a time span, that can be more or less precise:
+        `year[-month[-day]][:endYear[-endMonth[-endDay]]` (e.g. `1705-09-21` or `1705-09-23:1705-09-29`)
+        (Gregorian Calendar is always implied.)
+        :param str start_span: If the date is precise, this represents the date;
+        if the date is a time span, this represents the start of the time span.
+        :param str|False end_span: if the date is precise, this must be False;
+        else, this represents the end of the time span.
+        """
+        def __init__(self, start_span, end_span):
+            self.start_span = start_span
+            self.end_span = end_span
+
+        def get_standardized_date_string(self):
+            """
+            Returns a string representation of the standardized date.
+            :return str:
+            """
+            res = self.start_span
+            if self.end_span:
+                res = res + u':' + self.end_span
```

Comment:
You  could use a constant on class level for a separator like `:`.

Reply:
https://github.com/dhlab-basel/alephmarcreader/blob/2ed60c1f1fb7bc7999afb09c63b223f009ccea87/alephmarcreader/abstractalephmarcreader.py#L110

https://github.com/dhlab-basel/alephmarcreader/blob/2ed60c1f1fb7bc7999afb09c63b223f009ccea87/alephmarcreader/abstractalephmarcreader.py#L133

Is that how you imagined it?

Reply:
great, thx